Userland/Utilities/slugify.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2023, Gurkirat Singh <tbhaxor@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Error.h>
+#include <AK/Slugify.h>
+#include <AK/String.h>
+#include <AK/Vector.h>
+#include <LibCore/ArgsParser.h>
+#include <LibMain/Main.h>
+#include <LibUnicode/Normalize.h>
+
+ErrorOr<int> serenity_main(Main::Arguments arguments)
+{
+    Vector<String> inputs;
+    String output_type;
+    String glue;
+    bool spa;
+
+    Core::ArgsParser parser;
+    parser.set_general_help("Slugify is a simple text to slug transform utility\n$ slugify 'Serenity is a cool ### PROject123.'");
+    parser.add_option(output_type, "Output format to choose from 'md', 'html', 'plain'. (default: md)", "format", 'f', "FORMAT");
+    parser.add_option(glue, "Specify delimiter to join the parts. (default: -)", "glue", 'g', "GLUE");
+    parser.add_option(spa, "Prepends hash/pound symble (#) to the slugified string when set, otherwise slash (/). Useful for markdowns like in GitHub (default: false)", "single-page", 's');
+    parser.add_positional_argument(inputs, "Input strings to be slugified.", "inputs");
+    if (!parser.parse(arguments)) {
+        parser.print_usage(stderr, arguments.strings[0]);
+        return 1;
+    }
+
+    char cglue = glue.is_empty() ? '-' : static_cast<char>(*glue.code_points().begin());
```

Comment:
This will break if they pass non-ascii like `--glue "🦬"`, and act unexpectedly for multi-character strings like `--glue "hello"`.

I'd recommend making it a custom Option (`parser.add_option({ .option_name = "..." ...});`) with a check that it's a single ascii character, and making the `glue` variable a `char` instead of a `String`.

Reply:
Ohk we can also use char. It is implemented in the ArgumentParser 

https://github.com/SerenityOS/serenity/blob/867f7da0179c6b77c729247f16e403c37b0322fa/Userland/Libraries/LibCore/ArgsParser.h#L121-L122

Reply:
[Char is integral type](https://en.cppreference.com/w/cpp/types/is_integral). I can test it

Reply:
I get it now what you meant. 😅

Reply:
It is fixed now. Please https://github.com/SerenityOS/serenity/pull/21245/commits/86a1b242b23c9352537d05e4f2a2707820a927ae